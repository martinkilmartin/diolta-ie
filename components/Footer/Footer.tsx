import { Link } from '../Link'
import { Copyright } from '../Copyright'

type Props = {
  links?: string[]
  brand: string
}

const Footer = ({ links, brand }: Props): JSX.Element => (
  <footer className="w-full max-w-screen-xl px-4 py-8 mx-auto bg-neutral text-neutral-content rounded-box">
    {links && (
      <ul className="flex flex-wrap justify-between max-w-screen-md mx-auto text-lg font-light">
        {links.map((link, index) => (
          <Link text={link} key={`${link.substring(0, 4)}_${index}`} />
        ))}
      </ul>
    )}
    <Copyright brand={brand} />
  </footer>
)

export default Footer
