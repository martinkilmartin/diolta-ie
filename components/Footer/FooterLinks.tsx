import Link from 'next/link'

type Props = {
  links: {
    title: string
    urls: {
      path: string
      text: string
    }[]
  }[]
}

const FooterLinks = ({ links }: Props): JSX.Element => (
  <div className="grid-rows footer">
    {links.map((link, i) => (
      <div key={i}>
        <span className="footer-title">{link.title}</span>
        {link.urls.map((link, index) => (
          <Link key={index} href={link.path}>
            <a className="link link-hover">{link.text}</a>
          </Link>
        ))}
      </div>
    ))}
  </div>
)

export default FooterLinks
