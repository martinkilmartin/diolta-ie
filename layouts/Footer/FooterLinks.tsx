type Props = {
  links: {
    title: string
    urls: {
      path: string
      text: string
    }[]
  }[]
}

const FooterLinks = ({ links }: Props): JSX.Element => (
  <div className="grid-rows footer">
    {links.map((link, i) => (
      <div key={i}>
        <span className="opacity-100 footer-title">{link.title}</span>
        {link.urls.map((link, index) => (
          <a key={index} href={link.path} className="link link-hover">
            {link.text}
          </a>
        ))}
      </div>
    ))}
  </div>
)

export default FooterLinks
