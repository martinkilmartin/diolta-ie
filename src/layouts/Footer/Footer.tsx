import { FooterLinks } from '@layouts/Footer'
import { Copyright } from '@components/Copyright'

import { FOOTER_LINKS } from '@constants/FOOTER_LINKS'

const Footer = (): JSX.Element => (
  <footer className="p-10 footer bg-primary text-primary-content footer-center">
    <FooterLinks links={FOOTER_LINKS} />
    <div>
      <Copyright />
    </div>
  </footer>
)

export default Footer
