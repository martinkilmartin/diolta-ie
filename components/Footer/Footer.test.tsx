import renderer from 'react-test-renderer'
import Footer from './Footer'

test('Footer displays copyright with current year and brand', () => {
  const brand = 'Acme Teo.'
  const footerRenderer = renderer.create(<Footer brand={brand} />)
  const tree = footerRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const footerInstance = footerRenderer.root
  const footerElement = footerInstance.findByProps({
    className:
      'w-full max-w-screen-xl px-4 py-8 mx-auto bg-primary text-primary-content rounded-box',
  })
  expect(
    footerElement.findByProps({
      className:
        'flex items-center justify-center pt-10 font-light text-center sm:pt-12',
    }).children[0]
  ).toEqual(`© ${new Date().getFullYear()} ${brand}`)
})

test('Footer displays links and copyright with current year and brand', () => {
  const footerLinks = ['abcdefgh', 'ijklmn', 'opq', 'rst', 'uvwxyz']
  const brand = 'Acme Teo.'
  const footerRenderer = renderer.create(
    <Footer brand={brand} links={footerLinks} />
  )
  const tree = footerRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const footerInstance = footerRenderer.root
  const footerElement = footerInstance.findByProps({
    className:
      'w-full max-w-screen-xl px-4 py-8 mx-auto bg-primary text-primary-content rounded-box',
  })
  expect(
    footerElement.findByProps({
      className:
        'flex items-center justify-center pt-10 font-light text-center sm:pt-12',
    }).children[0]
  ).toEqual(`© ${new Date().getFullYear()} ${brand}`)
})
