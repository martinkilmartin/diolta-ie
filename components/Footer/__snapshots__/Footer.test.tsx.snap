// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer displays copyright with current year and brand 1`] = `
<footer
  className="w-full max-w-screen-xl px-4 py-8 mx-auto bg-neutral text-neutral-content rounded-box"
>
  <div
    className="flex items-center justify-center pt-10 font-light text-center sm:pt-12"
  >
    © 2021 Acme Teo.
  </div>
</footer>
`;

exports[`Footer displays links and copyright with current year and brand 1`] = `
<footer
  className="w-full max-w-screen-xl px-4 py-8 mx-auto bg-neutral text-neutral-content rounded-box"
>
  <ul
    className="flex flex-wrap justify-between max-w-screen-md mx-auto text-lg font-light"
  >
    <li
      className="my-2"
    >
      <a
        className="transition-colors duration-200 hover:text-neutral-focus"
        href="#"
      >
        abcdefgh
      </a>
    </li>
    <li
      className="my-2"
    >
      <a
        className="transition-colors duration-200 hover:text-neutral-focus"
        href="#"
      >
        ijklmn
      </a>
    </li>
    <li
      className="my-2"
    >
      <a
        className="transition-colors duration-200 hover:text-neutral-focus"
        href="#"
      >
        opq
      </a>
    </li>
    <li
      className="my-2"
    >
      <a
        className="transition-colors duration-200 hover:text-neutral-focus"
        href="#"
      >
        rst
      </a>
    </li>
    <li
      className="my-2"
    >
      <a
        className="transition-colors duration-200 hover:text-neutral-focus"
        href="#"
      >
        uvwxyz
      </a>
    </li>
  </ul>
  <div
    className="flex items-center justify-center pt-10 font-light text-center sm:pt-12"
  >
    © 2021 Acme Teo.
  </div>
</footer>
`;
